test/integration/inherit-env-blacklist-existing.sh

```diff
@@ -46,4 +46,32 @@ if [[ $(vault_token_present) -eq 0 ]]; then
 else
     echo "not ok 4 - VAULT_TOKEN present while blacklisted via config file"
 fi
+
+# Test that the blacklist from config can be disabled with the CLI
+vault_token_present_with_disabled_blacklist() {
+    stack exec --cwd $TEMP_RUN_DIRECTORY -- vaultenv \
+        --no-connect-tls \
+        --host ${VAULT_HOST} \
+        --port ${VAULT_PORT} \
+        --secrets-file ${VAULT_SEEDS} \
+        --inherit-env-blacklist '' \
+        "$@" \
+        /usr/bin/env | grep --count "VAULT_TOKEN=integration"
+}
+
+if [[ $(vault_token_present_with_disabled_blacklist) -eq 1 ]]; then
```

Comment:
Can you think of a way to invert this test condition (check for `0` exit code)?

Currently, this test would also pass if you pass in a wrong CLI argument or something else goes wrong, e.g. when you change `--port` to `--prt`

Reply:
> Currently, this test would also pass if you pass in a wrong CLI argument or something else goes wrong, e.g. when you change `--port` to `--prt`

It does not: `grep` outputs the total amount of matches when invoked with `--count`, and by using `$(...)` `test` compares the output of the command to `1`, instead of the exit code.

<details>

<summary>I also tested this</summary>

```
$ git diff | cat
diff --git a/test/integration/inherit-env-blacklist-existing.sh b/test/integration/inherit-env-blacklist-existing.sh
index 37bf13d..ba2d0f6 100755
--- a/test/integration/inherit-env-blacklist-existing.sh
+++ b/test/integration/inherit-env-blacklist-existing.sh
@@ -52,7 +52,7 @@ vault_token_present_with_disabled_blacklist() {
     stack exec --cwd $TEMP_RUN_DIRECTORY -- vaultenv \
         --no-connect-tls \
         --host ${VAULT_HOST} \
-        --port ${VAULT_PORT} \
+        --prt ${VAULT_PORT} \
         --secrets-file ${VAULT_SEEDS} \
         --inherit-env-blacklist '' \
         "$@" \
$ ./integration_test.sh
Running tests for Key/Value mount V1
Success! Disabled the secrets engine (if it existed) at: secret/
Success! Enabled the kv secrets engine at: secret/
Running all tests
NOTE: You might see errors in the log. That's part of what we test.
Look at the test summary report to see if there's anything wrong.
integration/happy_path.sh ...................... ok
integration/environment_inherritance.sh ........ ok
integration/inherit-env-blacklist-both.sh ...... ok
integration/duplicate_env_var.sh ............... [ERROR] Found duplicate environment variable "TESTING_KEY"
integration/duplicate_env_var.sh ............... ok
integration/bad_token.sh ....................... [ERROR] Invalid Vault token
integration/bad_token.sh ....................... ok
integration/no_token_configured.sh ............. [ERROR] Invalid Vault token
integration/no_token_configured.sh ............. ok
integration/inherit-env-blacklist-existing.sh .. 1/6 Invalid option `--prt'

Usage: vaultenv [--version] [--host HOST] [--port PORT] [--addr ADDR]
                [--token TOKEN] [--secrets-file FILENAME] [CMD] [ARGS...]
                ([--no-connect-tls] | [--connect-tls]) ([--no-validate-certs] |
                [--validate-certs]) ([--no-inherit-env] | [--inherit-env])
                [--inherit-env-blacklist COMMA_SEPARATED_NAMES]
                [--retry-base-delay-milliseconds MILLISECONDS]
                [--retry-attempts NUM] [--log-level error | info] [--use-path]
integration/inherit-env-blacklist-existing.sh .. Failed 1/6 subtests
integration/inherit-env-blacklist-secret.sh .... ok
integration/invalid_addr.sh .................... 1/5 [ERROR] Confliciting configuration values in cli options
Vault address: "http://localhost:8200"
Vault host: "localhost"
Vault port: 8200
Use TLS: True
Hint: This can happen when you provide a `addr` which conflicts with `port`, `host`, or `[no-]connect-tls` in either the environment variables or the CLI.


[ERROR] Confliciting configuration values in cli options
Vault address: "https://localhost:8200"
Vault host: "localhost"
Vault port: 8200
Use TLS: False
Hint: This can happen when you provide a `addr` which conflicts with `port`, `host`, or `[no-]connect-tls` in either the environment variables or the CLI.


[ERROR] Confliciting configuration values in cli options
Vault address: "http://localhost:8200"
Vault host: "invalidhost"
Vault port: 8200
Use TLS: False
Hint: This can happen when you provide a `addr` which conflicts with `port`, `host`, or `[no-]connect-tls` in either the environment variables or the CLI.


[ERROR] Confliciting configuration values in cli options
Vault address: "http://localhost:8200"
Vault host: "localhost"
Vault port: 4321
Use TLS: False
Hint: This can happen when you provide a `addr` which conflicts with `port`, `host`, or `[no-]connect-tls` in either the environment variables or the CLI.


integration/invalid_addr.sh .................... ok

Test Summary Report
-------------------
integration/inherit-env-blacklist-existing.sh (Wstat: 0 Tests: 6 Failed: 1)
  Failed test:  5
Files=9, Tests=31,  5 wallclock secs ( 0.03 usr  0.01 sys +  4.07 cusr  0.34 csys =  4.45 CPU)
Result: FAIL
```

</details>

Reply:
Ah, I was too quick to comment! Then this looks good to me